sdks/python/apache_beam/runners/runner.py

```diff
@@ -119,6 +119,19 @@ class PipelineRunner(object):
 
   def run(self, transform, options=None):
     """Run the given transform or callable with this runner.
+
+    Blocks until the pipeline is complete.  See also `PipelineRunner.run_async`.
+    """
+    result = self.run_async(transform, options)
+    result.wait_until_finish()
```

Comment:
fn_api_runner's RunnerResult.wait_until_finish() doesn't seem to actually wait.

Reply:
That's because run() doesn't actually run asynchronously.